src/types.d.ts

```diff
@@ -17,6 +17,7 @@ export interface GrafanaQuery extends DataQuery {
   alias?: string;
   target?: string;
   payload: string;
+  label?: string;
```

Comment:
Can you explain why is the label needed?

Reply:
I'd say we need it to work properly with map responses[^1].

Whenever we fetch such a map for the first time, we get an object where `target` is a value and `label` is a text. If we only keep `value` in the JSON, then we need to compare the `value` to the map list to obtain the `text` part of it on reload. I also think if different `text` entries have same values, it might be potentially incorrect as well. I haven't tested the latter case, though.

To avoid these regular calculations, we could just keep both fields (`text` and `value`) in the state so it works for both cases - maps and arrays:

- With map scenarios we show `text` value in the dropdown list, while sending `value` value to the backend;
- With array scenarios we don't really care as they are the same.

[^1]: `{ "text": "upper_25", "value": 1}`